### DISCUSSION TITLE
game boarden hexagons an counters

### DISCUSSION THREAD

**Discussion by aviejo** [boardgameio/boardgame.io#1103] [General]
Hello, I am looking for a tool to make a game that supports board in hesagnos and counter. that the counters are placed on the hexagons or on other counters. is it possible with this framework?

**Comment by cwatsonc** [comment]
Aviejo, other developers have implemented hexagonal game layouts on this framework, they identify each location as part of a grid.  This platform is for using websocket communications for client-server games.  The board, and its state are modeled on the server, the clients connect and manipulate the server object, and those changes are reflected remotely on all connected clients.  See the game created by https://github.com/lkingsford (Emu Bay) which uses a hexagonal layout using bgio.

**Comment by jonaswre** [comment]
I'm not sure if this is what you are asking but
in a hexagonal game the tiles are identified by x,y,z (even when they are only in 2D)

> ... The insight is that a uniform hex grid is the isometric projection of an infinite grid of unit cubes whose centers satisfy the equation x+y+z=0. Thus, hairy problems with hexes in 2-D become nicer problems with cubes in 3-D in very much the same way that using homogenous coordinates linearizes projective transformations. ...

http://www-cs-students.stanford.edu/~amitp/Articles/Hexagon2.html

This helped me creating a hexagonal game.

**Comment by delucis** [comment]
https://github.com/nicolodavis/bgio-react-component-toolkit might also be of interest here — no longer actively maintained but this contains some example components for hex grids.

**Comment by aviejo** [comment]
![image](https://user-images.githubusercontent.com/42932426/197346554-77939ede-61ea-4994-818f-5aaca52f8f13.png)
Thanks for the info. The game idea is of this type according to the attached image.
